fastchat/constants.py

```diff
@@ -37,6 +38,7 @@
 WORKER_API_EMBEDDING_BATCH_SIZE = int(
     os.getenv("FASTCHAT_WORKER_API_EMBEDDING_BATCH_SIZE", 4)
 )
+HTTPX_TIMEOUT = os.getenv("FASTCHAT_HTTPX_TIMEOUT", DEFAULT_TIMEOUT_CONFIG)
```

Comment:
why do we need this?

Reply:
I met `httpx` timeout due to its `Async client` api, and discussions at this issue [Increase httpx timeout to prevent requests from erroring out](https://github.com/gradio-app/gradio/issues/5143) has provided a way to solve this problem by modifying the `httpx` _config.py line 370 which is in /path/to/python_lib/site-packages/httpx. I thought that is too aggressive so I add an environmental parameter. It is valued that the problem comes out only when network condition are `bad`, but I can't always reproduce this problem. It seems to be resolved in `gradio` newest release version using requests as a replace. I recommend to apply `HTTPX_TIMEOUT` argument because even `gradio` official can't reproduce this but it actually exists, users can use this argument to increase httpx_timeout in case they meet again.
![httpx_timeout_config](https://github.com/lm-sys/FastChat/assets/98650340/7b55e2ab-203a-4f13-a5e1-3206eb3906bc)